public/app/features/alerting/unified/hooks/useCombinedRuleNamespaces.ts

```diff
@@ -88,9 +88,12 @@ export function useCombinedRuleNamespaces(): CombinedRuleNamespace[] {
             }
 
             (group.rules ?? []).forEach((rule) => {
-              const existingRule = combinedGroup!.rules.find((existingRule) => {
-                return !existingRule.promRule && isCombinedRuleEqualToPromRule(existingRule, rule);
-              });
+              //@TODO remove ?? once api is fixed
+              const existingRule = isGrafanaRulesSource(rulesSource)
```

Comment:
nit: maybe break this out into a helper function `getExistingRule`.

Reply:
- [ ] break it up